_posts/2020-07-12-ray-tune.md

```diff
@@ -0,0 +1,457 @@
+> stopping, distributed execution) can provide significant speedups over grid
+> search and random search.
+>
+> However, the machine learning ecosystem is missing a solution that provides
+> users with the ability to leverage these new algorithms while allowing users
+> to stay within the Scikit-Learn API. In this blog post, we introduce
+> tune-sklearn [Ray's tuning library] to bridge this gap. Tune-sklearn is a
+> drop-in replacement for Scikit-Learn’s model selection module with
+> state-of-the-art optimization features.
+>
+> —[GridSearchCV 2.0 — New and Improved][1]
+
+This claim is inaccurate: for over a year Dask-ML has provided access to
+"cutting edge hyperparameter tuning techniques" with a Scikit-learn compatible
+API. To correct their statement, let's see how Dask-ML compares to each of the
+features that tune-sklearn provides:
+
+[dml-hod]:https://ml.dask.org/hyper-parameter-search.html
+
+> Here’s what [Ray's] tune-sklearn has to offer:
+> 1. **Consistency with Scikit-Learn API** ...
+> 2. **Modern hyperparameter tuning techniques** ...
+> 3. **Framework support** ...
+> 4. **Scale up** ...
+>
+> [Ray's] Tune-sklearn is also **fast**.
+
+Dask-ML's model selection module has every one of the features. Let's look at
+each feature and see what Dask-ML and Ray have to offer:
+
+### Consistency with the Scikit-learn API
+
+*Dask-ML is also consistent with the Scikit-learn API.*
+
+Here's how to use Scikit-learn's, Dask-ML's and Ray's tune-sklearn
+hyperparameter optimization:
+
+``` python
+from scipy.stats import loguniform
+from sklearn.datasets import make_classification
+from sklearn.linear_model import SGDClassifier
+from dask.distributed import Client
+client = Client()
+
+X, y = make_classification()
+model = SGDClassifier()
+params = {"loss": ["hinge", "log", "perceptron"],
+          "alpha": loguniform(1e-5, 1e-3)}
+
+# Scikit-learn hyperparameter optimization
+from sklearn.model_selection import RandomizedSearchCV
+search = RandomizedSearchCV(clf, params, ...)
+search.fit(X, y)
+
+# Dask hyperparameter optimization
+from dask_ml.model_selection import HyperbandSearchCV
+search = HyperbandSearchCV(model, params, ...)
+search.fit(X, y, classes=[0, 1])
+
+# Ray hyperparameter optimization
+from tune_sklearn import TuneSearchCV
+search = TuneSearchCV(clf, params, ...)
+search.fit(X, y, classes=[0, 1])
+```
+
+Clearly, both Dask-ML and Ray's tune-sklearn are Scikit-learn compatible. Now
+let's focus on how each search performs and how it's configured.
+
+[skl-eg]:https://scikit-learn.org/stable/auto_examples/model_selection/plot_randomized_search.html#sphx-glr-auto-examples-model-selection-plot-randomized-search-py
+[sgd-eg]:https://github.com/ray-project/tune-sklearn/blob/31f228e21ef632a89a74947252d8ad5323cbd043/examples/sgd.py
+
+### Modern hypereparameter tuning techniques
+
+Ray's tune-sklearn provides access to advanced hyperparameter optimization, and
```

Comment:
The previous section started with the note in italics. Let's be consistent.

Reply:
Thanks, that removes the word "also" from each claim.